container/trie/sparse_merkle.go

```diff
@@ -197,6 +197,20 @@ func VerifyMerkleBranch(root, item []byte, merkleIndex int, proof [][]byte, dept
 	return bytes.Equal(root, node[:])
 }
 
+// VerifyMerkleProof given a trie root, a leaf, the generalized merkle index
+// of the leaf in the trie, and the proof itself.
+func VerifyMerkleProof(root, item []byte, merkleIndex int, proof [][]byte) bool {
```

Comment:
We don't need a depth to verify merkle proofs for the state, but need a depth for verifying deposit merkle proofs, so I separate the methods